src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -583,12 +590,21 @@ namespace winrt::TerminalApp::implementation
 
         else
         {
-            auto conhostConn = TerminalConnection::ConptyConnection(settings.Commandline(),
-                                                                    settings.StartingDirectory(),
-                                                                    settings.StartingTitle(),
-                                                                    settings.InitialRows(),
-                                                                    settings.InitialCols(),
-                                                                    winrt::guid());
+            std::wstring guidWString = Utils::GuidToString(profileGuid);
+
+            StringMap envMap{};
+            envMap.Insert(L"WT_PROFILE_ID", guidWString);
+            envMap.Insert(L"WSLENV", L"WT_PROFILE_ID");
```

Comment:
Minor concern here: WSLENV could exist in the user's preexisting environment, and we would stomp it by doing this; is there a way for us to get/append the current value like we do in ConptyConnection @ 113?

Reply:
Good catch. Fixed.